pna.p4

```diff
@@ -545,6 +545,224 @@ extern Digest<T> {
 }
 // END:Digest_extern
 
+// BEGIN:ExactMap_extern
+struct exactmap_const_entry_t<K,V> {
+    K key;
+    V val;
+}
+
+struct exactmap_initial_entry_t<K,V> {
+    bool const_entry;
+    K key;
+    V val;
+}
+
+/**
+ * The type K must be a struct type.  Each of its members becomes a
```

Comment:
K as struct type: It is possible for user to pass any type for this template parameter. There is no restrictions on number of fields, nesting the struct type. Hope, it is up to the target to restrict on these semantics?

Reply:
Yes, targets can limit the total size and/or type of fields within the struct type K, e.g. if the user wants a 10,000 bit key, that is too much for most targets.